matrix-sdk/src/main/java/org/matrix/androidsdk/data/cryptostore/MXFileCryptoStore.java

```diff
@@ -392,21 +393,24 @@ private boolean storeObject(Object object, File file, String description) {
                 }
 
                 FileOutputStream fos = new FileOutputStream(file);
-                CipherOutputStream cos = CompatUtil.createCipherOutputStream(fos, mContext);
+                GZIPOutputStream gz;
 
-                if (cos != null) {
-                    GZIPOutputStream gz = CompatUtil.createGzipOutputStream(cos);
-                    ObjectOutputStream out = new ObjectOutputStream(gz);
-
-                    out.writeObject(object);
-                    out.flush();
-                    out.close();
-
-                    succeed = true;
-                    Log.d(LOG_TAG, "## storeObject () : " + description + " done in " + (System.currentTimeMillis() - t0) + " ms");
+                if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.KITKAT) {
+                    CipherOutputStream cos = CompatUtil.createCipherOutputStream(fos, mContext);
```

Comment:
It's strange to test API level before calling a function from `CompatUtil`.
Is it possible that `CompatUtil.createCipherOutputStream()` return type is returning an `OutputStream` instead of a `CipherOutputStream` (just changing the type to a more generic type), and for SDK_INT < KITKAT, just return the unmodified for parameter (the value of `fos` in this case)